src/components/PressableProps.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  ColorValue,
+  AccessibilityProps,
+  ViewProps,
+  NativeSyntheticEvent,
+  TargetedEvent,
+  Insets,
+  StyleProp,
+  ViewStyle,
+} from 'react-native';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+} from '../handlers/gestureHandlerCommon';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+import { LongPressGestureHandlerEventPayload } from '../handlers/LongPressGestureHandler';
+
+export interface PressableStateCallbackType {
+  readonly pressed: boolean;
+}
+
+export interface PressableAndroidRippleConfig {
+  color?: null | ColorValue | undefined;
+  borderless?: null | boolean | undefined;
+  radius?: null | number | undefined;
+  foreground?: null | boolean | undefined;
+}
+
+export interface PressableProps
+  extends AccessibilityProps,
+    Omit<ViewProps, 'children' | 'style' | 'hitSlop'> {
+  /**
+   * Called when the hover is activated to provide visual feedback.
+   */
+  onHoverIn?:
+    | null
+    | ((
+        event: GestureStateChangeEvent<HoverGestureHandlerEventPayload>
+      ) => void);
+
+  /**
+   * Called when the hover is deactivated to undo visual feedback.
+   */
+  onHoverOut?:
+    | null
+    | ((
+        event: GestureStateChangeEvent<HoverGestureHandlerEventPayload>
+      ) => void);
+
+  /**
+   * Called when a single tap gesture is detected.
+   */
+  onPress?: null | ((event: GestureTouchEvent) => void);
+
+  /**
+   * Called when a touch is engaged before `onPress`.
+   */
+  onPressIn?: null | ((event: GestureTouchEvent) => void);
+
+  /**
+   * Called when a touch is released before `onPress`.
+   */
+  onPressOut?: null | ((event: GestureTouchEvent) => void);
+
+  /**
+   * Called when a long-tap gesture is detected.
+   */
+  onLongPress?:
+    | null
+    | ((
+        event: GestureStateChangeEvent<LongPressGestureHandlerEventPayload>
+      ) => void);
```

Comment:
I think that here we would like to match the event types from the RN's implementation.